lgc/state/PipelineState.cpp

```diff
@@ -1121,7 +1121,63 @@ unsigned PipelineState::getShaderWaveSize(ShaderStage stage) {
   assert(stage <= ShaderStageCompute);
   if (!m_waveSize[stage])
     setShaderDefaultWaveSize(stage);
-  return m_waveSize[stage];
+
+  unsigned waveSize = m_waveSize[stage];
+
+  if (getTargetInfo().getGfxIpVersion().major >= 9) {
+    waveSize = getMergedShaderWaveSize(stage);
+  }
+
+  return waveSize;
+}
+
+// =====================================================================================================================
+// Gets wave size for the merged shader stage
+//
+// NOTE: For GFX9+, two shaders are merged as a shader pair. The wave size is determined by the larger one.
+//
+// @param stage : Shader stage
+unsigned PipelineState::getMergedShaderWaveSize(ShaderStage stage) {
+  unsigned waveSize = m_waveSize[stage];
```

Comment:
It is better to add a assert here to make sure we are GFX9+ like this:

assert(gfxip >= 9);